mirascope/base/types.py

```diff
@@ -1,11 +1,46 @@
 """Types for working with Mirascope prompts."""
-from typing import Callable, Iterable, Literal, Optional, Type, TypedDict, Union
+from enum import Enum
+from inspect import isclass
+from typing import (
+    Annotated,
+    Any,
+    Callable,
+    Iterable,
+    Literal,
+    Optional,
+    Type,
+    TypedDict,
+    Union,
+    get_origin,
+)
 
 from pydantic import BaseModel
 from typing_extensions import Required
 
 from .tools import BaseTool
 
+BaseType = Union[
+    str,
+    int,
+    float,
+    bool,
+    list,
+    set,
+    tuple,
```

Comment:
what about dict? or bytes?

Reply:
not handled. dict just doesn't work, bytes is also hard and not necessarily that useful, so I've excluded.

We can look to add them later, but it wasn't a simple addition